src/App.js

```diff
@@ -27,11 +28,19 @@ function App() {
           <Route
             exact
             path="/list"
-            element={<ListView token={token} setToken={setToken} />}
+            element={
+              <RequireToken>
+                <ListView token={token} setToken={setToken} />
+              </RequireToken>
+            }
           />
           <Route
             path="/add"
-            element={<AddItemView token={token} setToken={setToken} />}
+            element={
+              <RequireToken>
+                <AddItemView token={token} setToken={setToken} />
+              </RequireToken>
+            }
```

Comment:
This was cool to see you wrap the AddItemView with the newly created RequireToken component. **Smart thinking.**